### ISSUE TITLE
`brew upgrade curl` failed in the configure step

### ISSUE THREAD

**Issue by kprav33n** [Linuxbrew/homebrew-core#9347]
I did a routine `brew update` followed by `brew upgrade`. `curl` failed to upgrade. It failed in the configure step with the following error message.

```
configure: error: C compiler cannot create executables
See `config.log' for more details
```

The relevant section in `config.log` is the following.

```
configure:4247: checking whether the C compiler works
configure:4269: gcc-5 -mmacosx-version-min=10.7   conftest.c  >&5
gcc-5: error: unrecognized command line option '-mmacosx-version-min=10.7'
```

[Output of `gist-logs --config --doctor curl`](https://gist.github.com/kprav33n/e3367979262b709f73a5f2047ef0ddf0)

I further examined the recipe using `brew edit curl`, and found the following line to cause problem.

```
ENV.append_to_cflags "-mmacosx-version-min=10.7" if MacOS.version <= :lion
```

Upon commenting this line out, and running `brew upgrade curl`, installation was successful.

The issue seemed to be introduced by fac4a57dcd3d82a003f32579fa6a85fa04ce1103. I don't know enough to fix this in a clean way. Otherwise, I could create a PR.

**Comment by iMichka** [comment]
Thanks for reporting this. I opened a PR to fix this: #9349

**Comment by iMichka** [comment]
This should be fixed now. Sorry for the disagreement, I did not notice that line during the last merges today.
